src/xlsx.js

```diff
@@ -0,0 +1,99 @@
+export class ExcelWorkbook {
+  constructor(workbook) {
+    Object.defineProperty(this, "_", {value: workbook});
+  }
+  sheetNames() {
+    return this._.worksheets.map((sheet) => sheet.name);
+  }
+  sheet(name, {range, headers = false} = {}) {
+    const sheet = this._.getWorksheet(
+      typeof name === "number" ? this.sheetNames()[name] : name + ""
+    );
+    if (!sheet) throw new Error(`Sheet not found: ${name}`);
+    return extract(sheet, {range, headers});
+  }
+}
+
+function extract(sheet, {range, headers}) {
+  let [[c0, r0], [c1, r1]] = parseRange(range, sheet);
+  const seen = new Set();
+  const names = [];
+  const headerRow = headers && sheet._rows[r0++];
+  function name(n) {
+    if (!names[n]) {
+      let name = (headerRow ? valueOf(headerRow._cells[n]) : AA(n)) || AA(n);
+      while (seen.has(name)) name += "_";
+      seen.add((names[n] = name));
+    }
+    return names[n];
+  }
+  if (headerRow) for (let c = c0; c <= c1; c++) name(c);
+
+  const output = new Array(r1 - r0 + 1).fill({});
```

Comment:
This is filling the output with a shared empty object for all rows, whereas the rows with values are reassigned below to new objects. Do we want to use an empty object to represent rows without values, rather than undefined? If we do want to use an empty object, I think we’ll still want a distinct object for each row, rather than sharing the object across rows. That could be done by moving the `output[r - r0] = {}` below before the `continue` rather that using array.fill here.

Reply:
I think I actually tried to simply use sparseness initially but `Inputs.table` had trouble with it? re-using the same object was a pre-optimization from a memory usage standpoint. I also toyed with using a `Symbol(empty)` to make it even more explicit.

I slightly want to auto-filter these rows out of the return value since I feel like in usage that would be one of the first things I'd always end up writing anyway in the notebook, but that seemed like possibly surprising behavior at the same time?